bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTableRegionLocator.java

```diff
@@ -64,7 +57,45 @@ public TableName getName() {
   }
 
   @Override
-  public CompletableFuture<HRegionLocation> getRegionLocation(byte[] arg0, boolean arg1) {
-    throw new UnsupportedOperationException("getRegionLocation"); // TODO
+  public CompletableFuture<HRegionLocation> getRegionLocation(byte[] row, boolean reload){
+    try {
+      this.regionsFuture = getRegionsAsync(reload);
+    } catch (IOException e) {
+      //As getRegionLocation() method is not throwing any exception hence needs to handle following way.
+      e.getCause();
```

Comment:
Please `return FutureUtils.failedFuture(e)` here.  `getRegionsAsync()` does have some logic to throw an IOException.  Or, even better,  `getRegionsAsync()` should not throw an exception, and should do its own wrapping of the execution with a `Futures.immediateFailFuture()`